packages/astro/src/core/build/generate.ts

```diff
@@ -59,7 +59,7 @@ function shouldSkipDraft(pageModule: ComponentInstance, astroConfig: AstroConfig
 		!astroConfig.markdown.drafts &&
 		// This is a draft post
 		'frontmatter' in pageModule &&
-		(pageModule as any).frontmatter.draft === true
+		(pageModule as any).frontmatter?.draft === true
```

Comment:
remark-mdx-frontmatter applies `export const frontmatter = undefined` when no frontmatter is present. This handles that case!